execution/aggregate/khashaggregate.go

```diff
@@ -205,41 +211,112 @@ func (a *kAggregate) init(ctx context.Context) error {
 	return nil
 }
 
-func (a *kAggregate) aggregate(t int64, result *[]model.StepVector, k int, sampleIDs []uint64, samples []float64) {
-	for i, sId := range sampleIDs {
-		h := a.inputToHeap[sId]
-		switch {
-		case h.Len() < k:
-			heap.Push(h, &entry{sId: sId, total: samples[i]})
+func (a *kAggregate) aggregate(t int64, result *[]model.StepVector, k int, sampleIDs []uint64, samples []float64, histogramIDs []uint64, histograms []*histogram.FloatHistogram) {
```

Comment:
This is handling a few cases now, I'd add a comment, detailing those for posterity.